src/annotator/notebook.js

```diff
@@ -29,26 +31,39 @@ export default class Notebook extends Delegator {
     super(element, config);
     this.frame = null;
 
+    this.groupId = null;
+    this.prevGroupId = null;
+
     this.container = document.createElement('div');
     this.container.style.display = 'none';
     this.container.className = 'notebook-outer';
 
-    this.subscribe('showNotebook', () => this.show());
+    this.subscribe('showNotebook', groupId => {
+      this.groupId = groupId;
+      this.show();
+    });
     this.subscribe('hideNotebook', () => this.hide());
     // If the sidebar has opened, get out of the way
     this.subscribe('sidebarOpened', () => this.hide());
   }
 
-  init() {
-    if (!this.frame) {
-      this.frame = createNotebookFrame(this.options);
+  update() {
+    // Create a new iFrame if we don't have one at all yet, or if the
+    // groupId has changed since last use
+    const needIframe =
+      !this.frame || !this.prevGroupId || this.prevGroupId !== this.groupId;
+    this.prevGroupId = this.groupId;
```

Comment:
My understanding is that this is just a temporary mechanism and we expect to change it once there is a direct communication path between the notebook and sidebar?

I suppose we could perhaps have gone even simpler here and just recreated the notebook every time you opened it until that mechanism was ready.

Reply:
> I suppose we could perhaps have gone even simpler here and just recreated the notebook every time you opened it until that mechanism was ready.

True, though if you don't change focused groups between Notebook "opens," the current implementation has the nice side effect of retaining state.